source/AI.cpp

```diff
@@ -5024,3 +5043,57 @@ void AI::UpdateOrders(const Ship &ship)
 		order.targetSystem = ship.GetSystem();
 	}
 }
+
+
+
+// Mission NPC blocks may define specific travel plans.
+void AI::IssueNPCOrders(Ship &ship, const System *targetSystem,
+	const map<const Planet *, bool> &stopovers, const Planet *destination)
+{
+	Orders newOrders;
+	const System *from = ship.GetSystem();
+	if(targetSystem)
+	{
+		RoutePlan routePlan(ship, *targetSystem, nullptr);
+		if(!routePlan.HasRoute())
+			ship.EraseWaypoint(targetSystem);
+		else
+		{
+			newOrders.type = Orders::TRAVEL_TO;
+			newOrders.targetSystem = targetSystem;
+			if(from == targetSystem)
+			{
+				// Travel to the next waypoint, if it exists.
+				ship.SetTargetStellar(nullptr);
+				const System *nextSystem = ship.NextWaypoint();
+				if(nextSystem)
+					newOrders.targetSystem = nextSystem;
+				else
+					newOrders.targetSystem = nullptr;
+			}
+		}
+	}
+
+	// If one of the planets in this system is a destination or stopover, it
+	// supercedes the order to travel to the next waypoint (unless already visited).
+	if(destination && destination->IsInSystem(from))
+	{
+		newOrders.type = Orders::LAND_ON;
+		newOrders.targetPlanet = destination;
+	}
+
+	for(const auto &it : stopovers)
+		if(!it.second && it.first->IsInSystem(from))
+		{
+			newOrders.type = Orders::LAND_ON;
+			newOrders.targetPlanet = it.first;
+			break;
+		}
```

Comment:
This seems to assume that the final destination and any stopovers will be along the path the ship chooses to take between the waypoints, but I don't think this requirement was ever enforced by `NPC`? This is going to be especially problematic for NPCs using stopovers selected by filters where it may not be possible to ensure that the systems appear in the waypoints list.
Is this handled elsewhere? Otherwise, it appears very easy for a ship to just never even try to visit some or all of its stopovers.

Reply:
Stopovers aren't supposed to tell the ship which planets to fly to, just where it needs to land. This is so that you could have, for example, and escort mission where the ship lands at its destination without you and then you go somewhere else.

Reply:
I could add a personality tag or something that adds all stopover systems to the waypoint list, but that might be best done as a separate PR since we're nearing the end of this cycle? Not sure.

Reply:
I think this is fine for now. NPCs can have waypoints and stopovers, and having both of them should allow content creators to combine them when needed.